assets/src/scripts/post-list-filters.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Toggles the post list filter container.
+ *
+ * @param {HTMLElement} filterButton - The button used to toggle the filter container.
+ * @param {HTMLElement} filterContainer - The filter container.
+ */
+const toggleFilterContainer = ( filterButton, filterContainer ) => {
+	/**
+	 * Click event handler for the filter button.
+	 */
+	const clickHandler = () => {
+		const isHidden = ! filterContainer.classList.contains( 'post-list-filter__container--open' );
+		filterContainer.classList.toggle( 'post-list-filter__container--open' );
+
+		filterButton.innerHTML = isHidden ? 'Hide filters' : filterButton.dataset.showFiltersAppliedButtonLabel;
+	};
+
+	filterButton.addEventListener( 'click', clickHandler );
+};
+
+/**
+ * Resets the date filters.
+ *
+ * @param {HTMLElement} fromDateInput - The input element for the "from" date.
+ * @param {HTMLElement} toDateInput - The input element for the "to" date.
+ */
+const resetDateFilters = ( fromDateInput, toDateInput ) => {
+	fromDateInput.value = '';
+	toDateInput.value = '';
+};
+
+/**
+ * Walk through the form elements and reset them.
+ *
+ * @param {HTMLElement} form - The form element.
+ * @returns {void}
+ */
+const resetFormFields = form => {
+
+	// Reset all inputs on form.
+	const inputs = form.querySelectorAll( 'input[type="text"], input[type="date"]' );
```

Comment:
This includes the search query field - is that intentional?

Reply:
Yes, that's the action of button "Clean filters" to guarantee that the form is being properly cleaned up entirely before submitting. Different of the action of the button "Reset" which reset only date filters.

Reply:
@MiguelAxcar Please flag this one for Greg in testing, in my own experimentation I found that I didn't expect this field to be cleared when I clicked this. I'd recommend seeing what @varnent feels is best.